argcomplete/shellintegration.py

```diff
@@ -55,8 +55,12 @@ def shellcode(executable, use_defaults=True, shell='bash', complete_arguments=No
         complete_options = " ".join(complete_arguments)
 
     if shell == 'bash':
-        code = bashcode
+        quoted_executables = ['"%s"' % i for i in executables]
```

Comment:
Should use `shlex.quote` for this (it's small enough that you could copy and paste it for use on Python 2).